app/webpacker/components/CompetitionsOverview/CompetitionsView.js

```diff
@@ -38,8 +38,14 @@ function CompetitionsView() {
   const [displayMode, setDisplayMode] = useState(() => getDisplayMode(searchParams));
   const [shouldShowRegStatus, setShouldShowRegStatus] = useState(false);
   const competitionQueryKey = useMemo(
-    () => calculateQueryKey(debouncedFilterState),
-    [debouncedFilterState],
+    () => calculateQueryKey(debouncedFilterState, canViewAdminDetails),
+    [debouncedFilterState, canViewAdminDetails],
+  );
+
+  // Need to make sure that people don't "hijack" admin mode by manipulating the URL
+  const shouldShowAdminDetails = useMemo(
+    () => canViewAdminDetails && filterState.shouldShowAdminDetails,
+    [canViewAdminDetails, filterState.shouldShowAdminDetails],
   );
```

Comment:
There's no benefit to memoizing a simple calculation resulting in a boolean.
```suggestion
  const shouldShowAdminDetails = canViewAdminDetails && filterState.shouldShowAdminDetails;
```